test-utils/docs/README.md

```diff
@@ -0,0 +1,156 @@
+
+The configuration files are located in the `./test-utils` folder. Alongside the standard configurations, we have included a `./test-utils/custom-environment.ts` file. This file addresses the issue of missing `window` methods, like `matchMedia`, in the testing environment.
+
+Additionally, we store all shared tests in this folder. There are two categories of shared tests. The first is `common.tsx`, which consists of tests that can be utilized independently in multiple components. The second is `{component}.tsx`, which includes tests specific to a particular component that can be reused in other components. For instance, `./src/components/BackgroundImage/BackgroundImage.tsx` utilizes `./src/components/Image/Image.tsx` as a child. Consequently, many tests for the `Image` component can be utilized to test the `BackgroundImage` component as well.
+
+✅ The shared tests of the second type are named after the component that possesses a lower level of abstraction.
+
+For instance, since the `Image` component is a part of the `BackgroundImage` component, the shared test should be named `image.tsx` rather than `background-image.tsx`.
+
+### Component Test
+
+All tests for a component are stored in the `__tests__` folder. Typically, there is a single `*.test.tsx` file within the folder, named after the component. However, there is no strict rule that restricts the number of test files that can be present.
+
+```
+...
+
+|- BackgroundImage
+  |- __stories__
+  |- __tests__
+    |- BackgroundImage.test.tsx
+  |- BackgroundImage.tsx
+  |- BackgroundImage.scss
+
+...
+```
+
+Typically, when working with `jest` tests, it's common to organize them into suites. These suites serve as containers for a group of related tests. Essentially, a suite is contained within a single file and brings together all the tests associated with a specific component.
+
+Follow the steps below to write your own test:
+
+- Use the describe function from `Jest` to define your test suites.
+- Inside each suite, use the test or it function to define your test cases.
+- Render your component using the `render` function.
+- Use the `screen` object from `@testing-library/react` to query and interact with the rendered component.
+- Use the `expect` function from `@testing-library/jest-dom` to make assertions about the rendered component.
+
+```ts
+describe('BackgroundImage', () => {
+    test('Render BackgroundImage by default', async () => {
+        render(<BackgroundImage qa={qaId} />);
+
+        const component = screen.getByTestId(qaId);
+        expect(component).toBeInTheDocument();
+        expect(component).toBeVisible();
+    });
+
+    ...
+```
+
+### Rules
+
+#### Avoid Direct Node Access
+
+It is important to avoid direct Node access when writing tests. This means refraining from using methods like `querySelector` or `querySelectorAll`. Instead, we should rely on testing library methods such as `getByRole`, `getByText`, or `getByTestId` to access components.
```

Comment:
Why do we want to avoid document.querySelector()?
As the [RTL philosophy](https://testing-library.com/docs/guiding-principles) suggests, we want to avoid selecting elements by classNames, ids, or even test ids, because that is not how the user would find the element on the screen.

querySelector as a query strategy is vulnerable to DOM changes or if we are basing the selectors on CSS Styling, the CSS Styling locators can change independently of the functionality and we may find functional-based tests failing due to styling changes.

https://medium.com/smallcase-engineering/why-should-we-not-use-document-queryselector-while-testing-react-components-835bbf34169e

Reply:
Can we add a brief explanation to Readme of why exactly this is important t?